module/vein_detection.py

```diff
@@ -59,22 +59,30 @@ def __clahe(self, frame, amount=1):
     # The parameter kernel_size is used to determine the amount of pixels used for the kernel in the median blur
     # A larger kernel means more pixels are used in the calculation but the image could become more blurry
     # Returns frame with less noise
-    def __image_denoising(self, frame, kernel_size=5):
+    def image_denoising(self, frame, kernel_size=5):
         """ Image denoising, remove noise from supplied image
         :param frame: frame obtained from the camera
         :param kernel_size: the amount of pixels used for the kernel in the median blur
         :return: frame with median blur applied
         """
-        frame = cv2.medianBlur(frame, kernel_size)
+        return cv2.medianBlur(frame, kernel_size)
 
-        return frame
-
-    def __adaptive_thresholding(self, frame):
+    # Adaptive thresholding is used to create a black/white image in which the veins can be clearly distinguished
+    # from the skin.
+    # Our implementation of adaptive thresholding uses Otsu's binarization.
+    # With Otsu's binarization the threshold in an image gets calculated so that the thresholding works on every image.
+    # Instead of getting the best threshold by trial and error Otsu's method can calculate the best threshold.
+    # The parameter 0 is the threshold value. This value is left at 0 because Otsu is used to calculate the threshold.
+    # The parameter 255 is the gray value the parts of the image get when they are above the threshold.
+    # THRESH_BINARY makes everything under the threshold black (0)
+    # and everything above the threshold the parameter color in this case 255.
+    # THRESH_OTSU is used to calculate the best threshold value.
```

Comment:
This text could be moved inside in the function's docstring.

Reply:
What would be the added value of this? The functions already have a docstring included.

Reply:
Its included in the generated Pydoc documentation if it's inside the docstring. Otherwise people (next year) will have to dig through the code to find this rather extensive comment.

Reply:
Fair enough @YouriOkkerse @denizlektemur can you format your descriptions to docstring?

Reply:
@nicovanbentum Fixed.